library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandlerTest.kt

```diff
@@ -0,0 +1,455 @@
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.inOrder
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.spy
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class RecordedDataQueueHandlerTest {
+    lateinit var testedHandler: RecordedDataQueueHandler
+
+    @Mock
+    lateinit var mockProcessor: RecordedDataProcessor
+
+    @Mock
+    lateinit var mockRumContextDataHandler: RumContextDataHandler
+
+    lateinit var mockExecutorService: ExecutorService
+
+    @Mock
+    lateinit var mockSystemInformation: SystemInformation
+
+    @Mock
+    lateinit var mockTimeProvider: SessionReplayTimeProvider
+
+    @Forgery
+    lateinit var fakeRumContextData: RumContextData
+
+    @Forgery
+    lateinit var fakeSnapshotQueueItem: SnapshotRecordedDataQueueItem
+
+    @Forgery
+    lateinit var fakeTouchQueueItem: TouchEventRecordedDataQueueItem
+
+    lateinit var fakeTouchData: List<MobileSegment.MobileRecord>
+
+    lateinit var fakeNodeData: List<Node>
+
+    @BeforeEach
+    fun setup(forge: Forge) {
+        mockExecutorService = spy(
+            ThreadPoolExecutor(
+                1,
+                10,
+                100,
+                TimeUnit.MILLISECONDS,
+                LinkedBlockingDeque()
+            )
+        )
+
+        fakeTouchData = forge.aList { mock() }
+
+        fakeNodeData = forge.aList { mock() }
+
+        testedHandler = RecordedDataQueueHandler(
+            mockProcessor,
+            mockRumContextDataHandler,
+            mockExecutorService,
+            mockTimeProvider
+        )
+    }
+
+    @Test
+    fun `M use executorService W tryToConsumeItems() { queue has items }`() {
+        // Given
+        testedHandler.recordedDataQueue.offer(fakeSnapshotQueueItem)
```

Comment:
why not using the public methods here ? 
`addSnapshotItem` and `addTouchEventItem` ?

Reply:
I think you need a test for each

Reply:
I've now added tests for each and moved to using the internal methods